neevaco/neeva-android
# Switch `NeevaUserToken` to be a `SessionToken`

* Better align how the `login` cookie works with the `preview` and `incognito` cookies by making the `NeevaUserToken` a `LoginToken` that inherits from `SessionToken`.
  * Allow `SessionToken#requestNewCookie` to be overridden and make it be a no-op for the `LoginToken` because login cookies are provided only when the user has signed in.

* Make `WebLayerModel` observe when the login cookie has changed and do whatever it needs to instead of having each callsite inconsistently call `onAuthTokenUpdated`.

* Fix the `CookieChangedCallback` so that we ask the `Browser` for the latest value of a cookie when it changes.  WebLayer seemingly incorrectly provides us the old value of a cookie instead of the updated version when the cookie changes, meaning that when we clear the user's cookies, we don't get alerted with the right value.

* Make the `RegularBrowserWrapper` keep track of when the login cookie changes and make it fetch updated user data instead of `NeevaActivity`.

* Fix when the app determines that the user has signed in before by confirming that the login cookie isn't empty.

## Checklists

### How was this tested?

### Manual test cases
Clearing page cookies:
1. Sign into the app
2. Confirm the user is signed in via the app Settings
3. Navigate to neeva.com and confirm that the user is signed in
4. Click on the page info button from the overflow menu
5. Click on "cookies" in the popup and delete them all
6. Confirm that the webpage reloads and sends the user to neeva's sign in page
7. Confirm that the user is not signed in via the app Settings

Clearing history:
1. Sign into the app
2. Confirm the user is signed in via the app Settings
3. Navigate to neeva.com and confirm that the user is signed in
4. Open History and clear all browsing data from all time
5. Navigate to neeva.com
6. Confirm the user is still signed in
7. Confirm that the user is still signed in via the app Settings

## Issues addressed
* Fixes https://github.com/neevaco/neeva-android/issues/987
* Fixes https://github.com/neevaco/neeva-android/issues/781
